compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -466,7 +466,14 @@ macro_rules! define_queries {
         }
 
         impl<'tcx> QueryDescription<QueryCtxt<'tcx>> for queries::$name<'tcx> {
-            rustc_query_description! { $name }
+            fn describe(tcx: QueryCtxt<'tcx>, key: Self::Key) -> String {
+                ::rustc_middle::query::descs::$name(tcx.tcx, key)
+            }
+
+            #[inline]
+            fn cache_on_disk(tcx: TyCtxt<'tcx>, key: &Self::Key) -> bool {
```

Comment:
`cache_on_disk` is typically very small and never used in `rustc_query_system`.  I'd like to remove it from the trait too, if possible.  It's used in `try_load_from_on_disk_cache`, so it's a bit more complex as a refactor.

Reply:
Doing this in a not-cursed way (I have several cursed ideas involving conjuring up generic fn defs using `transmute(())`) requires a few more refactors than I'd like to do here, so I'd prefer doing that in a follow-up PR.